optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/common/nearby/NearbyRandomFactory.java

```diff
@@ -0,0 +1,107 @@
+                .getNearbySelectionDistributionType() == NearbySelectionDistributionType.LINEAR_DISTRIBUTION
+                || nearbySelectionConfig.getLinearDistributionSizeMaximum() != null;
+        boolean parabolicDistributionEnabled = nearbySelectionConfig
+                .getNearbySelectionDistributionType() == NearbySelectionDistributionType.PARABOLIC_DISTRIBUTION
+                || nearbySelectionConfig.getParabolicDistributionSizeMaximum() != null;
+        boolean betaDistributionEnabled =
+                nearbySelectionConfig.getNearbySelectionDistributionType() == NearbySelectionDistributionType.BETA_DISTRIBUTION
+                        || nearbySelectionConfig.getBetaDistributionAlpha() != null
+                        || nearbySelectionConfig.getBetaDistributionBeta() != null;
+        if (!randomSelection) {
+            if (blockDistributionEnabled || linearDistributionEnabled || parabolicDistributionEnabled
+                    || betaDistributionEnabled) {
+                throw new IllegalArgumentException("The nearbySelectorConfig (" + nearbySelectionConfig
+                        + ") with randomSelection (" + randomSelection
+                        + ") has distribution parameters.");
+            }
+            return null;
+        }
+        if (blockDistributionEnabled && linearDistributionEnabled) {
+            throw new IllegalArgumentException("The nearbySelectorConfig (" + nearbySelectionConfig
+                    + ") has both blockDistribution and linearDistribution parameters.");
+        }
+        if (blockDistributionEnabled && parabolicDistributionEnabled) {
+            throw new IllegalArgumentException("The nearbySelectorConfig (" + nearbySelectionConfig
+                    + ") has both blockDistribution and parabolicDistribution parameters.");
+        }
+        if (blockDistributionEnabled && betaDistributionEnabled) {
+            throw new IllegalArgumentException("The nearbySelectorConfig (" + nearbySelectionConfig
+                    + ") has both blockDistribution and betaDistribution parameters.");
+        }
+        if (linearDistributionEnabled && parabolicDistributionEnabled) {
+            throw new IllegalArgumentException("The nearbySelectorConfig (" + nearbySelectionConfig
+                    + ") has both linearDistribution and parabolicDistribution parameters.");
+        }
+        if (linearDistributionEnabled && betaDistributionEnabled) {
+            throw new IllegalArgumentException("The nearbySelectorConfig (" + nearbySelectionConfig
+                    + ") has both linearDistribution and betaDistribution parameters.");
+        }
+        if (parabolicDistributionEnabled && betaDistributionEnabled) {
+            throw new IllegalArgumentException("The nearbySelectorConfig (" + nearbySelectionConfig
+                    + ") has both parabolicDistribution and betaDistribution parameters.");
+        }
+        if (blockDistributionEnabled) {
```

Comment:
Personally, I'd make this part of the `switch`, too. There should only be one place where decisions are made from a single piece of data.